### ISSUE TITLE
SnmpEngine() autogenerated snmpengineid differs on windows and linux

### ISSUE THREAD

**Issue by durdin85** [pysnmp/pysnmp#52]
Hi, I've noticed the code
```
SnmpEngine().snmpEngineID.prettyPrint()
```
as run on Windows Python would actually generate shorter SNMPEngineID (20 characters) than on Linux python instance (34 characters). Taking into consideration, that first characters are always same representing type, company id, and format used, there's very little of remaining "random" bytes left, making possibility of generating duplicate ID (maybe?) on windows.

I haven't been able to figure out how the engineid is generated, but would it be nice to have consistent ID's generated on any platform. 
Same problem is present also in the original code [etingof](https://github.com/etingof)/[pysnmp](https://github.com/etingof/pysnmp)

**Comment by lextm** [comment]
PySNMP is very Linux/UNIX centric, so in many places it works badly on Windows. What you observed is due the exception on line 222,

https://github.com/etingof/pysnmp/blob/master/pysnmp/smi/mibs/SNMP-FRAMEWORK-MIB.py#L222

where on Windows `os.uname()` won't work.

But still there is randomness in the generated ID and I don't think it will easily lead to duplication.

**Comment by lextm** [comment]
Fixed in release 5.0.28 of `pysnmp-lextudio`.

**Comment by durdin85** [comment]
> Fixed in release 5.0.28 of `pysnmp-lextudio`, https://pypi.org/project/pysnmp-lextudio/

Thank you for quick response. I just checked the updated file on: https://github.com/lextudio/pysnmp/blob/4a271c84b024e637e9252a2526c8201d8d0ea329/pysnmp/smi/mibs/SNMP-FRAMEWORK-MIB.py#L52

And I can confirm it now makes longer SnmpEngineID also on Windows. But I wonder if it's actually not too long now (in my case 40 characters). RFC says max 32 octets, but seems I now faced problem that some systems are interpreting that as 32 hexadecimal/textual characters... but this is for another bug report on different project(s) ;)
